test/e2e/notifier/update/update.go

```diff
@@ -0,0 +1,207 @@
+				assert.Equal(t, test.ExpectedWebhookPayload.EventStatus, lastSeenPayload.EventStatus)
+				assert.Equal(t, test.ExpectedWebhookPayload.Summary, lastSeenPayload.Summary)
+			}
+		})
+	}
+	//resetting update operation to false
+	utils.AllowedUpdateEventsMap[utils.KindNS{Resource: "v1/pods", Namespace: "all"}] = config.UpdateSetting{Fields: []string(nil), IncludeDiff: false}
+	utils.AllowedEventKindsMap[utils.EventKind{Resource: "v1/pods", Namespace: "all", EventType: "update"}] = false
+}
+
+// Run tests
+func (c *context) Run(t *testing.T) {
+	t.Run("update resource", c.testUpdateResource)
+	t.Run("skip update event", c.testSKipUpdateEvent)
+}
+
+// E2ETests runs create notification tests
+func E2ETests(testEnv *env.TestEnv) env.E2ETest {
+	return &context{
+		testEnv,
+	}
+}
+
+func (c *context) testSKipUpdateEvent(t *testing.T) {
+	//Modifying AllowedEventKindsMap configure dummy namespace for update event and ignore all
+	utils.AllowedEventKindsMap[utils.EventKind{Resource: "v1/pods", Namespace: "dummy", EventType: "update"}] = true
+	utils.AllowedEventKindsMap[utils.EventKind{Resource: "v1/pods", Namespace: "all", EventType: "update"}] = false
+	//test scenarios
+	tests := map[string]testutils.UpdateObjects{
+		"skip update event for namespaces not configured": {
+			// update operation not allowed for Pod in test namespace so event should be skipped
+			GVR:       schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"},
+			Kind:      "Pod",
+			Namespace: "test",
+			Specs:     &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-pod-update"}, Spec: v1.PodSpec{Containers: []v1.Container{{Name: "test-pod-container", Image: "tomcat:9.0.34"}}}},
+		},
+		"skip update event for resources not added": {
+			// update operation not allowed for namespaces in test_config so event should be skipped
+			GVR:   schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"},
+			Kind:  "Namespace",
+			Specs: &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "abc"}}},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			//checking if update operation is true
+			resource := fmt.Sprintf("%s/%s/%s", test.GVR.Group, test.GVR.Version, test.GVR.Resource)
+			if test.GVR.Group == "" {
+				resource = fmt.Sprintf("%s/%s", test.GVR.Version, test.GVR.Resource)
+			}
+			isAllowed := utils.AllowedEventKindsMap[utils.EventKind{
+				Resource:  resource,
+				Namespace: "all",
+				EventType: config.UpdateEvent}] ||
+				utils.AllowedEventKindsMap[utils.EventKind{
+					Resource:  resource,
+					Namespace: test.Namespace,
+					EventType: config.UpdateEvent}]
+			assert.Equal(t, isAllowed, false)
+		})
+	}
+
+	//Resetting original configuration as per test_config
+	utils.AllowedEventKindsMap[utils.EventKind{Resource: "v1/pods", Namespace: "dummy", EventType: "update"}] = false
+	utils.AllowedEventKindsMap[utils.EventKind{Resource: "v1/pods", Namespace: "all", EventType: "update"}] = true
```

Comment:
We can just delete the keys with `defer delete(utils.AllowedEventKindsMap, eventKind)`